site/topics/genetic-operators/genetic-operators.rst

```diff
@@ -190,6 +198,207 @@ Creep Mutation
+
+
+* Many of the previously discussed genetic operators are problematic since they may destroy the permutation property
+
+
+Crossovers
+----------
+
+Order Crossover
+^^^^^^^^^^^^^^^
+
+* Select two indices randomly
+* Copy the elements between the selected indices to a child
+* Copy the missing elements in the child from the other chromosome in the order they appear after the second index
+* Best described with an example
+
+#. Select two indices at random
+
+    * Here, indices 3 and 7 are selected
+
+
+#. Copy the elements between the selected indices to a child chromosome
+
+    * Typically the larger index is not included in the copy
+    * Here, elements at indices 3, 4, 5, and 6 are copied
+
+    .. figure:: ../genetic-operators/order_crossover_1.png
+        :width: 600 px
+        :align: center
+
+        Copy the elements between the selected indices to a child. Only one child chromosome is shown here.
+
+
+#. Copy elements from the other parent to the child in the order they appear, starting at the larger index
+
+    * Wrap to index 0 where necessary
+    * Here, the copying would start at index 7, which contains the element 1
+    * The 1 would be copied to the child as it is not contained within the child chromosome
+    * The value at index 8 is a 4, but would not get copied since it already exists in the child
+    * The next index would be 0 as there is no index 9, which contains a 9, thus it is copied to the child
+    * etc.
+    * The values that are copied are 1, 9, 3, 8, and 2, in that order
+
+
+    .. figure:: ../genetic-operators/order_crossover_2.png
+        :width: 600 px
+        :align: center
+
+        Copy the elements from the other parent, in order, starting after the larger index. Only copy values that are not
+        already contained within the child.
+
+
+#. Repeat the same idea for the other child
+
+
+Partially Mapped Crossover
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* This one is rather complex and is tough to explain in words
+* Below are figures that help explain the process
+
+#. Select two indices at random
+
+    * Here, indices 3 and 7 are selected
+
+
+#. Copy the elements between the selected indices to a child chromosome
+
+    * Typically the larger index is not included in the copy
+    * Here, elements at indices 3, 4, 5, and 6 are copied
+
+    .. figure:: ../genetic-operators/partially_mapped_crossover_1.png
+        :width: 600 px
+        :align: center
+
+        Copy the elements between the selected indices to a child. Only one child chromosome is shown here.
+
+
+#. Starting at the first selected index in the other parent, copy non-copied elements in the child by
+
+    * Finding the index of the element that exists in the child at the index of the non-copied index
+    * Here, the value of 8 was not copied and exists at index 4
+    * The value in index 4 of the child is 4
+    * The value of 4 exists at index 8 in the parent
+    * This, the value of 8 is copied into index 8
```

Comment:
This might be simpler to define in math as two partial functions, one of which is recursive. Text here is a little overwhelming.